src/components/Tokens/TokenDetails/index.tsx

```diff
@@ -234,13 +229,13 @@ export default function TokenDetails({
 
         <RightPanel onClick={() => isBlockedToken && setOpenTokenSafetyModal(true)}>
           <div style={{ pointerEvents: isBlockedToken ? 'none' : 'auto' }}>
-            <Widget
-              defaultTokens={{
-                [Field.INPUT]: inputToken ?? undefined,
-                default: detailedToken ?? undefined,
+            <Swap
+              defaultChainId={pageChainId}
+              prefilledState={{
```

Comment:
hmm so the chain id should always be coupled with `currencyId` right? can we just amend `currencyId` to include chain like mobile does

`{chainId}-{address}` format

Reply:
actually, might make more sense to leave the currencyIds as is but use a `useEffect` internal to the swap component that handles chain switching

Reply:
we don't want to switch automatically, we want to show a button to switch chains in the swap component. 

as for combining the values, I worry that changing the format of a "Currency ID" in this one place would be confusing, since the ID is just the address everywhere else.

also, this "default" chain ID only applies to one token and doesn't really fit with the current approach of passing a subset of swap state into `prefilledState` so we'd have to change the API a bit. i'm leaning towards keeping this prop, but let me know what you think.

Reply:
shouldn't it just be a `chainId` prop then? default implies that it can change within the component internally, but seems like `chain` should be a fully controlled prop

Reply:
i renamed to `pageChainId` which i think is clearer